### DISCUSSION TITLE
UI enhancements - Some Suggestions:

### DISCUSSION THREAD

**Discussion by jarrodCoombes** [advplyr/audiobookshelf#149] [Ideas]
**Library Section**

The shelf view is nice, but it relies on the cover to be correct in order for you to ID the book. If the cover is wrong, you won't know until you go play the book. Also, some of the covers are super hard to read.

It would be nice if:

When you hover over the cover it tells you the book name (similar to when you hover over a series tile in that section).

There was an option to switch to a list mode, where you had the books listed in columns of: Cover, Author, Title, Series etc. Bonus points if hovering over the book cover makes it pop out in a larger format for review. With a check box in the first column it would become much easier to do bulk actions like set genres or add tags (select the ones you want, then click "Tag" type in the tag and hit "save" type action). This would also allow for easy checking that all your covers are correct.

**Series Section**

Currently the most visible cover in the series section is the last one in the series, this should probably be the first one instead as most series seem to be named after the first book (at least of mine seem to be).

It would be nice to customize the series tiles with their own "covers" for the series. I am not too sure if this makes much sense, but it just came to mind while I was paging through that section.

Another crazy idea here would be when you hover over the series tile, the books fanned out and you could then click on one (play, edit, navigate to etc).

**Comment by advplyr** [comment]
I liked the fan covers idea so I tried it out, but I'm not sure where users should click to go to the series page. And where to display the series title. What do you think?

![seriescoverfan](https://user-images.githubusercontent.com/67830747/139354141-bffc95b8-351f-4f2f-a1ee-8734711551d6.gif)

**Reply by advplyr** [reply]
I just pushed `v1.5.7` that shows the first book in the series on top.

**Comment by jarrodCoombes** [comment]
Oh man, that is pretty damned awesome.

Maybe just add a "hidden" tile on the bottom of the stack that has the series title on it, and it fans out to the far right. That shows you the series title and clicking on it could take you to the series directly.

**Reply by advplyr** [reply]
Looks like it's missing something. Not sure.

![seriescoverfan_v2](https://user-images.githubusercontent.com/67830747/139509865-10d905fb-d30b-4077-a148-b4bd1ac7e16c.gif)

**Reply by jarrodCoombes** [reply]
Agreed, that looks off. I think the concept is good though. Can you make the series tile transparent (so just the name shows) and put it on top so it fans out to the left?

**Comment by advplyr** [comment]
The list view that you mentioned has been a long time coming as well, but I can't picture how this would look. I recently added a list view on the mobile app because it is especially important there, but I don't think it looks good.

It doesn't need to use the bookshelf background, but I would like it to look simple and clean. Any thoughts on a specific design? Some examples of software that does it right maybe.

![image](https://user-images.githubusercontent.com/67830747/139447689-cff3f44c-9b74-4b77-9d56-9b4458dce3e6.png)

**Comment by jarrodCoombes** [comment]
Personally I think that listview looks good, a nice balance being functional and using the real estate of the phone.

For the web app, it probably makes sense to start with what it is we'd be trying to solve with the list view. For me it would for a quick way to identify issue and mis-matches with the cover tile and the book (there were a few mis-matches that I did catch in the series view). So with that in mind I personally would see it is a utilitarian view, and therefore it does not necessarily need to be pretty, heck it could be something that just the admin gets to use. 

That said, I think you definitely should consider making the list view you have on the mobile an option on the web app, just with multiple books on each shelf. It completely solves the issue with some covers being hard to read, and could help ID mismatched cover tiles.

**Reply by advplyr** [reply]
Any suggestions on where to put the toggle button to change views?

**Comment by jarrodCoombes** [comment]
I am envisioning a button on the bottom left corner (in the same section that has the home, library etc buttons. Click it and sub menu come out the side, and you can select between whatever views there are.

Alternatively it could be at the top right, next to the sort drop down, or even the bottom right, just above the tile size selector.

**Reply by advplyr** [reply]
The thing is the list view will only be available on the library page. I think that is the only place it is really useful as well. So I think the top right makes the most sense, I will try it out.

**Reply by jarrodCoombes** [reply]
Ahh, yes, very good point, it does not need to be visible all the time.

**Reply by advplyr** [reply]
I set the max width of the rows to 1400px because it looked bad full width.
This is still missing details like genres, narrator, series and ebooks, but this was enough to get an idea of what it would look like.

Notice in the top right I used 2 icons grid & list that can be toggled between.

![image](https://user-images.githubusercontent.com/67830747/139514885-e1b92926-01e8-42ec-94ed-b4703fdedf4b.png)

Let me know what you think so far.

**Reply by jarrodCoombes** [reply]
Hmm, I like the left side, but not overly enamoured with the right. Thinking forget about the synopsis in this list, rather just have the tiles and the bit next to it, and the play showing when you hover over the cover. Something like this (excuse the bad paint editing).

![ListView](https://user-images.githubusercontent.com/58901318/139517021-5708075a-b26f-4d89-ba8f-4a8100c7ac37.png)

I guess at this point it's not so much a list view, as a detailed tile view.

**Reply by jarrodCoombes** [reply]
I think the list view should be a pretty standard thing, no graphics necessarily.

I am thinking this:
![image](https://user-images.githubusercontent.com/58901318/139517133-9d0c49ee-a282-43cd-a288-22d0b0edc0ae.png)

But with headers of Title, Author, Series, Year, Synopsis, ISBN, Genre, Tags etc. The idea being that it's mostly for identifying books missing data and/or incorrect data, you could click the headers to sort the each column.

The detailed tile list helps match covers with titles, and this list view helps locate books with missing data (and or mismatched text data, such as correct title wrong description).

**Reply by advplyr** [reply]
I redid the list view and released in `v1.5.9`. I would like to make the links clickable and show a larger cover image on hover. Let me know what you think so far.

![image](https://user-images.githubusercontent.com/67830747/139561403-92c28bfd-e143-4382-9163-f511d86cb37b.png)

**Reply by jarrodCoombes** [reply]
Looks great. I can easily tell which books have odd titles that need to be looked at, and I can easily tell what meta data is missing.

One bug I notice is if the title is long, it just runs into the next column, so you may need to do a carriage return after a certain number of characters there.

Hopefully you have not dropped the idea of the third view option, the "Detailed Tile view" as I personally like it as an option, and would probably use it as my default view.

**Reply by advplyr** [reply]
Just released [v1.6.0](https://github.com/advplyr/audiobookshelf/releases/tag/v1.6.0) fixed the title bug and moved the list view out of experimental features.
I think the details view should still be added, let's see the feedback from the list view.

**Reply by danneauxs** [reply]
Hi,
Whatever happened to this view?  This is exactly what I was looking for.  I haven't memorized all the books in a series that I haven't read and seeing this would let me read the description.  I think a hover mode to read over long names or descriptions that don't fit would be good.  I'd even be fine with a hover mode when viewing a series in just the cover view mode.  I actually came here looking for a discussion like this before starting one.

**Comment by advplyr** [comment]
Ok I just released `v1.5.8` where I put both the list view and the series hover fan in experimental.

Already noticed a bug with long titles and subtitles. A lot left to do on the list view.

![image](https://user-images.githubusercontent.com/67830747/139516030-00dd5631-2b33-4539-904e-cc195f8f5a0d.png)

**Comment by jarrodCoombes** [comment]
The list view is great! Very helpful for working out which books are missing information, I've already used it to quickly update the meta data for at least a dozen books. Coupled with the filtering it gets even better and easier to work through a large list of books.

**Comment by MidnightSnowleopard** [comment]
Is this list view available somewhere?  I haven't been able to find where to activate it with the icons to switch between views not available as they are in the screenshots here.

**Reply by advplyr** [reply]
I haven't gotten to re-building the list view since I implemented the lazy bookshelf. It isn't a big project, so you can expect it in the near future.

**Comment by szemlicka** [comment]
Hi,

I start working with this software and I like it. I search long time ago a client/server app like this.

Here are my comments for improvements...
Server Web Interface
1. Please sort the series. If you have a series like Perry Rhodan Silberedition, its not easy to find the next part. (found out that the "Volume" number is the part of the series. Is it possible to use the disc number in tag information as the series part?
2. For a huge library its not so easy to navigate. A shortcut at the top af each library,series or collection with an A - B- C .... can help

Android App
1. If I select a part of an series, I am not able to select the Author of the series. That works fine at the web interface.

**Reply by advplyr** [reply]
I'm not sure how common it is to use CD # as the volume #. My library has some audiobooks with CD # and none of them should be volume #. If this is a common thing then maybe a scanner setting could be added.
The scanner does look for Volume or Book in the book title and use that: https://www.audiobookshelf.org/docs#naming
and also uses the Series-Part ID3 tag to fill in the volume number.

I'm not sure what your idea for adding a shortcut at the top is meant to look like. Can you provide an example of what this looks like?

I confirmed the android app is still lacking the links on the audiobook pages.
